care/facility/api/viewsets/asset.py

```diff
@@ -188,10 +188,10 @@ def operate_assets(self, request, *args, **kwargs):
                 raise ValidationError({"asset_id": "is required"})
             if "action" not in request.data:
                 raise ValidationError({"action": "is required"})
-            asset_id = request.data["asset_id"]
             action = request.data["action"]
             asset: Asset = self.get_object()
-            result = AssetClasses(asset.asset_class).handle_action(action)
+            asset_class = AssetClasses[asset.asset_class].value(asset.meta)
```

Comment:
Handle unknown asset class in this method

Reply:
The error details are only returned when there is some validation error (post request body has missing/invalid parameters) with a status of 400. In the case of all other errors, only a message of "Internal server error" is returned with a status of 500. So, will there be any point in raising a separate new error? In any case, it is all inside a try-except block.

Reply:
ideally, internal server errors are reserved for errors that are unknown in nature. since these are known exceptions we try to handle as much as possible.